### PR TITLE
Configure account name's hostname component

### PR DESCRIPTION
**Pull request by matthieugusmini**

### CODE REVIEW

File: api/v1alpha1/grant_types.go
Comment (mmontes11) [inline api/v1alpha1/grant_types.go:88]: What about adding the `AccountName` method to the `Grant` for the sake of unification?

File: api/v1alpha1/user_types.go
Comment (mmontes11) [inline api/v1alpha1/user_types.go:72]: great 👍🏻

**Review by mmontes11 — CHANGES_REQUESTED** [review]
I think it would be worth adding `AccountName` method to `Grant` so it looks like `User`.  This way we can pass it directly here, as you did with `CreateUser` and `DropUser`: 
https://github.com/mariadb-operator/mariadb-operator/blob/f25831199002c637fdb67f258e333ce91e48f9c5/pkg/client/client.go#L220

Other than that, LGTM!

File: api/v1alpha1/grant_types.go
Comment (matthieugusmini) [reply api/v1alpha1/grant_types.go:88]: @mmontes11 Sure could do that but will also imply that we change the `GrantOpts` struct definition, replacing the `Username` and `Host` field by a `AccountName` string.

Also it seems like the `Privileges`, `Database`, `Table` and `AccountName` fields of the `GrantOpts` are always required. What do you think about changing the signature of the `Grant` and `Revoke` methods so they take these values as regular parameters and we would only keep the optional arguments of the option in the `GrantOpts` 🤔

Also does not seem like the [`REVOKE` statement](https://mariadb.com/kb/en/revoke/) takes any optional argument so what about dropping the `GrantOpts` from the signature? And if needed we later we could add a dedicated `RevokeOpts` struct 😄 

Something such as:
```go
func (c *Client) Grant(ctx context.Context, privileges []string, database, table, accountName string, opts GrantOpts) error

func (c *Client) Revoke(ctx context.Context, privileges []string, database, table, accountName string) error
```

Comment (mmontes11) [reply api/v1alpha1/grant_types.go:88]: In that case the `GrantOpts` would be just the `GrantOption` right?
```golang
type GrantOpts struct {
	GrantOption bool
}
```
We should still support revoking the `GRANT OPTION` so we would need to pass the `GrantOpts` to `Revoke` as well. For indicating that is nullable, we could pass it with a pointer, as follows:
```golang
func (c *Client) Grant(ctx context.Context, privileges []string, database, table, accountName string, opts *GrantOpts) error

func (c *Client) Revoke(ctx context.Context, privileges []string, database, table, accountName string,  opts *GrantOpts) error
```
What do you think?

Comment (matthieugusmini) [reply api/v1alpha1/grant_types.go:88]: Yes only the `GrantOption` will remain. Didn't take a closer look at the `GRANT` statement document but there's might be more support options by MariaDB itself though.

Passing a pointer to suggest the fact that the `GrantOpts` is optional sounds good. We could  also pass it as a [functional option](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) so other developers who don't need options when calling these methods doesn't even need to pass any value when calling it.

I'll push an implementation example to make it easier to visualize 😉

Comment (mmontes11) [reply api/v1alpha1/grant_types.go:88]: Agreed, let's go with an options pattern interface. You have an example here:
- https://github.com/mariadb-operator/mariadb-operator/blob/main/pkg/controller/galera/controller.go#L53

**Comment by mmontes11** [comment]
Some conflicts after merging this [PR](https://github.com/mariadb-operator/mariadb-operator/pull/178), sorry about that. Let me know if you have any questions!

File: pkg/client/client.go
Comment (mmontes11) [inline pkg/client/client.go:243]: awesome 🚀

**Comment by matthieugusmini** [comment]
@mmontes11 Yeah sure you can merge this PR if it looks good to you 👍🏼

File: pkg/client/client.go
Comment (mmontes11) [reply pkg/client/client.go:191]: > `client.Client`

This is the kubernetes client provided by `controller-runtime`

> What about renaming the package to something more explicit about its intent?

Agreed the naming is not great given the number of different clients we have. I would suggest `mariadbclient` to be more explicit. Feel free to raise another PR

**Review by mmontes11 — APPROVED** [review]

**Pull request merged**
